src/util/cert/libcrypto/cert.c

```diff
@@ -288,33 +373,84 @@ static errno_t ec_pub_key_to_ssh(TALLOC_CTX *mem_ctx, EVP_PKEY *cert_pub_key,
 #define SSH_RSA_HEADER "ssh-rsa"
 #define SSH_RSA_HEADER_LEN (sizeof(SSH_RSA_HEADER) - 1)
 
+static int sss_rsa_get_key(EVP_PKEY *cert_pub_key,
+                           BIGNUM **_n, BIGNUM **_e)
+{
+    int ret;
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+    *_n = NULL;
+    ret = EVP_PKEY_get_bn_param(cert_pub_key, OSSL_PKEY_PARAM_RSA_N, _n);
+    if (ret != 1) {
+        ret = EINVAL;
+        goto done;
+    }
+
+    *_e = NULL;
+    ret = EVP_PKEY_get_bn_param(cert_pub_key, OSSL_PKEY_PARAM_RSA_E, _e);
+    if (ret != 1) {
+        ret = EINVAL;
+        goto done;
+    }
+
+#else
+
+    const BIGNUM *tmp_n;
+    const BIGNUM *tmp_e:
+
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+    const RSA *rsa_pub_key = NULL;
+    rsa_pub_key = EVP_PKEY_get0_RSA(cert_pub_key);
+    if (rsa_pub_key == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    RSA_get0_key(rsa_pub_key, tmp_n, tmp_e, NULL);
+#else
```

Comment:
Do we really need to still support openssl older than 1.0.1?

Last 1.0.0 - t  - 2015

Reply:
Since there code was already there and is not complex I was just lazy and kept it to avoid any dedicated release note about the removed support. But in general I won't mind to drop the  support but maybe in a dedicate PR?

Reply:
ok